### PR TITLE
Add null check on avatar on chat page (#4175)

### PR DESCRIPTION
**Pull request by sunnyssr**
## What kind of change does this PR introduce?

This PR introduces a bug fix for chat which results in a crash

## What is the current behavior?

Currently, when one user who has sent a message on chats quit the chat, it results in  crash for all other users who are still active.
https://github.com/codesandbox/codesandbox-client/issues/4175

## What is the new behavior?

This is not a feature change. In new behaviour if users quit the app after sending a message in chat, it should continue working for other users.

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

1. Step A
2. Step B
3. Step C
Yes, I tested it by doing the chat. I used three devices(different browsers) to join the chat. Sent a message by one and quit.
## Checklist

- [ ] Documentation
- [x] Testing 
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/4176/commits/45784fa26e85cf31330a028f59c7d5f541d19334 is at https://pr4176.build.csb.dev/s/new.

**Comment by CompuIves** [comment]
Thanks a lot for the timely fix!! I've just taken a look, and it works well! It just looks a bit quirky: 

image

Maybe we should hide the Avatar altogether to make it look well? And I think we should also add ` (left)` to the name here: https://github.com/codesandbox/codesandbox-client/pull/4176/files#diff-7aad850f6f72d32dc15f29bcbfc7f34cR121. So we could do something like
`{name}{metadata ? '' : ' (left)'`.

**Comment by sunnyssr** [comment]
I think we should instead provide a fallback image which shows blank user image with a text saying left?

**Comment by CompuIves** [comment]
That's even better!! Together with the left text, I think that would be the best solution. @DannyRuchtie do you have an image for something like this?

**Comment by sunnyssr** [comment]
@CompuIves 
If we don't have image of the user, can we show CSB logo there?

**Comment by CompuIves** [comment]
I think it would be best if we show a user icon here if possible. @DannyRuchtie do you have an SVG to show here?

**Comment by DannyRuchtie** [comment]
User icon when there is no avatar and 20% opacity when they not there? 
The icon in the zip 

[chat.zip](https://github.com/codesandbox/codesandbox-client/files/4710269/chat.zip)

Screenshot 2020-06-01 at 10 30 48

**Comment by CompuIves** [comment]
Yeah that sounds great. We don't have the `avatarUrl` if the user isn't there right now, but I can add it in the future. For now I think we should show the user icon with 20% opacity.

**Comment by SaraVieira** [comment]
Design added 
CleanShot 2020-06-09 at 18 26 55


@CompuIves

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/Chat/index.tsx [hunk 1]
```diff
@@ -111,22 +126,30 @@ export const Chat: React.FC = () => {
...
-                      <Avatar
-                        color={color}
-                        alt={metadata.username}
-                        src={metadata.avatarUrl}
-                      />
+                      {metadata ? (
+                        <Avatar
+                          color={color}
+                          alt={metadata && metadata.username}
+                          src={metadata && metadata.avatarUrl}
```
Comment (CompuIves) [inline packages/app/src/app/pages/Sandbox/Editor/Workspace/Chat/index.tsx:137]:
```suggestion
                          alt={metadata.username}
                          src={metadata.avatarUrl}
```

**Comment by SaraVieira** [comment]
Thank you so much @sunnyssr !

**Pull request merged**